cluster/addons/iso-client/coreaffinity-isolator.go

```diff
@@ -24,21 +27,51 @@ const (
 	name = "iso"
 )
 
+// TODO: handle more than just unregistering  evenDispatcherClient
+func handleSIGTERM(sigterm chan os.Signal, client *aff.EventDispatcherClient, opaque *opaq.OpaqueIntegerResourceAdvertiser) {
```

Comment:
Could we rename this to `shutdownIsolator` or similar and call this from the TERM handler? There are a number of control flow paths in which we need to both unregister and unadvertise the OIRs (e.g. every instance of `glog.Fatalf(...)`).